Headers/aermass_container_mod.F90

```diff
@@ -0,0 +1,482 @@
+     REAL(fp), ALLOCATABLE :: SOAGX      (:,:,:)
+     REAL(fp), ALLOCATABLE :: PM25       (:,:,:)
+     REAL(fp), ALLOCATABLE :: PM10       (:,:,:)
+     REAL(fp), ALLOCATABLE :: ISOAAQ     (:,:,:)
+     REAL(fp), ALLOCATABLE :: SOAS       (:,:,:)
+     REAL(fp), ALLOCATABLE :: FRAC_SNA   (:,:,:,:)
+     REAL(fp), ALLOCATABLE :: DAERSL     (:,:,:,:)
+     REAL(fp), ALLOCATABLE :: WAERSL     (:,:,:,:)
+
+  END TYPE AerMassContainer
+!
+! !REMARKS:
+! 
+! !REVISION HISTORY:
+!  28 Mar 2023 - E. Lundgren- Initial version
+!  See https://github.com/geoschem/geos-chem for complete history
+!EOP
+!------------------------------------------------------------------------------
+!BOC
+!
+CONTAINS
+!EOC
+!------------------------------------------------------------------------------
+!                  GEOS-Chem Global Chemical Transport Model                  !
+!------------------------------------------------------------------------------
+!BOP
+!
+! !IROUTINE: Init_AerMass_Container
+!
+! !DESCRIPTION: Subroutine INIT\_AER\_Container allocates and initializes
+! the Aer container object.
+!\\
+!\\
+! !INTERFACE:
+!
+  SUBROUTINE Init_AerMass_Container( Input_Opt, State_Grid, Aer, RC )
+!
+! !USES:
+!
+    USE CMN_Size_Mod,   ONLY : NAER
+    USE Input_Opt_Mod,  ONLY : OptInput
+    USE State_Grid_Mod, ONLY : GrdState
+!
+! !INPUT PARAMETERS:
+!
+    TYPE(OptInput),      INTENT(IN)  :: Input_Opt  ! Input Options object
+    TYPE(GrdState),      INTENT(IN)  :: State_Grid ! Grid object
+!
+! !INPUT/OUTPUT PARAMETERS:
+!
+    TYPE(AerMassContainer),  POINTER :: Aer        ! Aerosol data container
+!
+! !OUTPUT PARAMETERS:
+!
+    INTEGER,             INTENT(OUT) :: RC         ! Success or failure?
+!
+! !REVISION HISTORY:
+!  28 Mar 2023 - E. Lundgren- Initial version
+!  See https://github.com/geoschem/geos-chem for complete history
+!EOP
+!------------------------------------------------------------------------------
+!BOC
+!
+! !LOCAL VARIABLES:
+!
+    CHARACTER(LEN=255) :: errMsg, thisLoc
+    INTEGER            :: NX, NY, NZ
+
+    !======================================================================
+    ! Init_AerMass_Container starts here
+    !======================================================================
+
+    ! Initialize local variables
+    RC = GC_SUCCESS
+    thisLoc = ' -> at Init_AerMass_Container (in module Headers/aermass_container_mod.F90)'
+    NX = State_Grid%NX
+    NY = State_Grid%NY
+    NZ = State_Grid%NZ
+
+    ! Exit immediately if this is a dry-run
+    IF ( Input_Opt%DryRun ) RETURN
+
+    !======================================================================
+    ! Initialize arrays
+    !======================================================================
+
+    ! OM/OC for POA
+    ALLOCATE( Aer%OCFPOA( NX, NY ), STAT=RC )
+    IF ( RC /= GC_SUCCESS ) THEN
+       errMsg = 'Error allocating array OCFPOA!'
+       CALL GC_Error( errMsg, RC, thisLoc )
+       RETURN
+    ENDIF
+    Aer%OCFPOA = 0.0_fp
+
```

Comment:
You can use the GC_CheckVar routine for convenience.  It'll print the error message and then set RC to failure if the array can't be allocated.
```F90
ALLOCATE( Aer%OCFPOA( NX, NY ), STAT=RC )
GC_CheckVar( 'OCFPOA', 0, RC )
IF ( RC /= GC_SUCCESS ) RETURN
Aer%OCFPOA = 0.0_fp
```

Reply:
Done. Thanks for the tip!